src/module_handle.cc

```diff
@@ -0,0 +1,345 @@
+		auto returnValue = mod->Evaluate(context_local);
+		Local<Value> moduleNamespace = mod->GetModuleNamespace();
+		global_namespace = std::make_shared<RemoteHandle<Value>>(moduleNamespace);
+		return returnValue;
+	}));
+}
+
+
+Local<Value> IsolatedModule::GetNamespace() {
+	Local<Object> value = ClassHandle::NewInstance<ReferenceHandle>(isolate, global_namespace, context_handle, ReferenceHandle::TypeOf::Object);
+	return value;
+}
+
+
+
+bool operator==(const IsolatedModule& isolated_module, const v8::Local<v8::Module>& mod) {
+	return isolated_module.module_handle->Deref() == mod;
+}
+
+ModuleHandle::ModuleHandleTransferable::ModuleHandleTransferable(shared_ptr<IsolateHolder> isolate, shared_ptr<IsolatedModule> isolated_module)
+	: isolate(std::move(isolate)), isolated_module(std::move(isolated_module))
+{
+}
+
+Local<Value> ModuleHandle::ModuleHandleTransferable::TransferIn() {
+	return ClassHandle::NewInstance<ModuleHandle>(isolate, isolated_module);
+};
+
+ModuleHandle::ModuleHandle(shared_ptr<IsolateHolder> isolate, shared_ptr<IsolatedModule> isolated_module)
+	: isolate(std::move(isolate)), isolated_module(std::move(isolated_module))
+{
+}
+
+Local<FunctionTemplate> ModuleHandle::Definition() {
+	return Inherit<TransferableHandle>(MakeClass(
+		"Module", nullptr,
+		"dependencySpecifiers", ParameterizeAccessor<decltype(&ModuleHandle::GetDependencySpecifiers), &ModuleHandle::GetDependencySpecifiers>(),
+		"setDependency", Parameterize<decltype(&ModuleHandle::SetDependency), &ModuleHandle::SetDependency>(),
+		"instantiate", Parameterize<decltype(&ModuleHandle::Instantiate<1>), &ModuleHandle::Instantiate<1>>(),
+		"instantiateSync", Parameterize<decltype(&ModuleHandle::Instantiate<0>), &ModuleHandle::Instantiate<0>>(),
+		"evaluate", Parameterize<decltype(&ModuleHandle::Evaluate<1>), &ModuleHandle::Evaluate<1>>(),
+		"evaluateSync", Parameterize<decltype(&ModuleHandle::Evaluate<0>), &ModuleHandle::Evaluate<0>>(),
+		"namespace", ParameterizeAccessor<decltype(&ModuleHandle::GetNamespace), &ModuleHandle::GetNamespace>()
+		//"getModuleNamespace", Parameterize<decltype(&ModuleHandle::GetModuleNamespace<1>), &ModuleHandle::GetModuleNamespace<1>>(),
+		//"getModuleNamespaceSync", Parameterize<decltype(&ModuleHandle::GetModuleNamespace<0>), &ModuleHandle::GetModuleNamespace<0>>()
+		/*"release", Parameterize<decltype(&ModuleHandle::Release), &ModuleHandle::Release>(),
+		*/
+	));
+}
+
+std::unique_ptr<Transferable> ModuleHandle::TransferOut() {
+	return std::make_unique<ModuleHandleTransferable>(isolate, isolated_module);
+}
+
+Local<Value> ModuleHandle::GetDependencySpecifiers() {
+	
+	std::lock_guard<IsolatedModule> lock(*isolated_module);
+	const std::vector<std::string> & dependencySpecifiers = isolated_module->GetDependencySpecifiers();
+	std::size_t size = dependencySpecifiers.size();
+
+	Local<Array> deps = Array::New(Isolate::GetCurrent(), size);
+
+	if (deps.IsEmpty()) {

```

Comment:
I just copy & paste from v8 guide that says it can fail: "https://github.com/v8/v8/wiki/Embedder's-Guide" but have removed it now. I assume we got bigger issues if it fails.